vector-im/element-android
# Reapply push rules on the decrypted event source (PSG-1146)

## Type of change


## Content

Reapply the local push rules on the decrypted event content.

## Motivation and context

When the notifications were enabled for `encrypted events`, we did not reapply the push rules on the decrypted content, so all the encrypted events were triggering a notification even if the notifications for `Direct Messages` or `Group Message` were enabled.

## Screenshots / GIFs





## Tests

- Enable all the default notifications from the app settings (DM, Group Messages, encrypted DM, encrypted Group Messages)
- Put the app in bg
- Send a message in encrypted DM & encrypted room (> 2 members) from another account
- Verify that you receive both notifications
- Disable DM, go in bg & send a message from the other account in both rooms
- Verify there is a single notification for group messages
- Enable DM & disable encrypted DM
- Test again and verify that you have the same result
- Do the same tests for group messages (you should only receive notifications in DM)

## Tested devices

- OS version(s):

## Checklist
